hs_core/hydroshare/resource.py

```diff
@@ -722,16 +725,32 @@ def delete_resource_file(pk, filename_or_id, user):
 
     return filename_or_id
 
+def get_resource_doi(res_id, pending):
+    doi_str = "http://dx.doi.org/10.4211/hs.{shortkey}".format(shortkey=res_id)
+    if pending:
+        return "{doi}pending".format(doi=doi_str)
+    else:
+        return doi_str
+
+
+def get_activated_doi(doi):
+    idx = doi.find('pending')
+    if idx >= 0:
+        return doi[:idx]
+    else:
+        return doi
 
-def publish_resource(pk):
+def publish_resource(user, pk):
```

Comment:
@dtarb This potential error we are catching here is purely CrossRef-related system error or potential network error, it has nothing to do with their metadata since we have already validated everything on our side before allowing them to publish. I believe this kind of potential error is very unlikely with CrossRef production system and a stable network connection, so I'd propose we go with a message similar to what I proposed above for this first pass implementation. If later on this becomes an issue, we can address it as a second pass improvement.